### DISCUSSION TITLE
How to insert a table with only one Row Type column?

### DISCUSSION THREAD

**Discussion by boson234** [trinodb/trino#16457] [Q&A] [answered]
steps:
1.  
CREATE TABLE hive.default.row_table (                           
    col1 ROW(col2 integer, col3 varchar)                         
 )   ;
2. insert into hive.default.row_table values (row(2, 'item'));
3. then the error as fellow:
 **Insert query has mismatched column types: Table: [row(col2 integer, col3 varchar)], Query: [integer, varchar(10)]**

as rules in SqlBase.g4 ,rowConstructor considers that **Row(1,'item1')**  equal to   **(1,'item1')**
    | '(' expression (',' expression)+ ')'                                                #rowConstructor
    | ROW '(' expression (',' expression)* ')'                                            #rowConstructor

if I want to Insert data into the table row_type, I have to write sql as fellow:
**insert into hive.default.row_table values (row(row(2, 'item')));**

that's not cool, so is there any better way to do this?

**Comment by ebyhr** [comment, marked as answer]
You can use `INSERT INTO ... SELECT ...` instead.

**Reply by tschaub** [reply]
@ebyhr (or anyone else) - I'm trying to `INSERT` large batches of data coming from Parquet files with group/struct columns.  

Using `ROW` types, it sounds like I would have to serialize this as `INSERT INTO ... SELECT ... UNION ALL SELECT ... UNION ALL SELECT ...` (and so on).  Are there performance tradeoffs to consider with this syntax vs `INSERT INTO ... VALUES ...`?

I'm trying to decide whether to represent this data using `ROW` type columns or whether to flatten each nested field into its own column.
